File: Linked List/src/206.reverse-linked-list.js
/*
 * @lc app=leetcode id=206 lang=javascript
 *
 * [206] Reverse Linked List
 */


// Definition for singly-linked list.
class ListNode {
  constructor(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
  }
}

// @lc code=start
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
  let newHead = new ListNode(0)
  
  let current = head;
  let temp;
  while (current !== null) {
    temp = newHead.next;
    newHead.next = new ListNode(current.val, temp);
    current = current.next;
  }

  return newHead.next;
};
// @lc code=end

File: Top Interview Questions/src/206.reverse-linked-list.java
/*
 * @lc app=leetcode id=206 lang=java
 *
 * [206] Reverse Linked List
 */

// Definition for singly-linked list.
class ListNode {
  int val;
  ListNode next;
  ListNode() {}
  ListNode(int val) { this.val = val; }
  ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}

// @lc code=start
class Solution {
  public ListNode reverseList(ListNode head) {
    ListNode newHead = null;
    
    while (head != null) {
      newHead = new ListNode(head.val, newHead);
      head = head.next;
    }

    return newHead;
  }
}
// @lc code=end

